test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1

```diff
@@ -0,0 +1,917 @@
+                Invoke-LanguageModeTestingSupportCmdlet -EnableConstrainedLanguageMode
+
+                $result = (& $scriptblock)
+            }
+            finally
+            {
+                Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode
+            }
+
+            ($result.Count -gt 0) | Should Be $true
+        }
+    }
+
+    Describe "Background jobs" -Tags 'Feature','RequireAdminOnWindows' {
+
+        Context "Background jobs in system lock down mode" {
+
+            # TODO
+            # Test is pending because PowerShell Core currently stubs out Windows system lockdown APIs
+            It "Verifies that background jobs in system lockdown mode run in constrained language" -Pending {
+
+                try
+                {
+                    Invoke-LanguageModeTestingSupportCmdlet -SetLockdownMode
+
+                    $job = Start-Job -ScriptBlock { [object]::Equals("A", "B") } | Wait-Job
+                    $expectedError = $job.ChildJobs[0].Error
+                    $job | Remove-Job
+                }
+                finally
+                {
+                    Invoke-LanguageModeTestingSupportCmdlet -RevertLockdownMode
+                }
+
+                $expectedError | Should Not Be $null
+                $expectedError.FullyQualifiedErrorId | Should Match "MethodInvocationNotSupportedInConstrainedLanguage"
+            }
+        }
+
+        Context "Background jobs within inconsistent mode" {
+
+            It "Verifies that background job is denied when mode is inconsistent" {
+
+                try
+                {
+                    Invoke-LanguageModeTestingSupportCmdlet -EnableConstrainedLanguageMode
+
+                    Start-Job { [object]::Equals("A", "B") }
+                    throw "No Exception!"
+                }
+                catch 
+                {
+                    $exception = $_
+                }
+                finally
+                {
+                    Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode
+                }
+
+                $exception.FullyQualifiedErrorId | Should Match "CannotStartJobInconsistentLanguageMode"
+            }
+        }
+    }
+
+    Describe "Add-Type in constrained language" -Tags 'Feature','RequireAdminOnWindows' {
+
+        It "Verifies Add-Type fails in constrained language mode" {
+
+            try
+            {
+                Invoke-LanguageModeTestingSupportCmdlet -EnableConstrainedLanguageMode
+
+                Add-Type -TypeDefinition 'public class ConstrainedLanguageTest { public static string Hello = "HelloConstrained"; }'
+                throw "No Exception!"
+            }
+            catch
+            {
+                $exception = $_
+            }
+            finally
+            {
+                Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode
+            }
+
+            $exception.FullyQualifiedErrorId | Should Match "CannotDefineNewType"

```

Comment:
Use ShouldBeErrorId

Reply:
Same reply.  BTW it is not necessary to add the same comment in each place :)

Reply:
I realized it after the Nth time, but early on it makes it easy to check if changes were applied in all cases :)